napari/_qt/containers/_base_item_view.py

```diff
@@ -118,7 +118,7 @@ def _sync_selection_models(self):
         for i in self._root.selection:
             idx = index_of(self.model(), i)
             selection.select(idx, idx)
-        sel_model.select(selection, sel_model.SelectionFlag.ClearAndSelect)
+        sel_model.select(selection, QItemSelectionModel.ClearAndSelect)
```

Comment:
for example, here old code is equivalent for 

```
QItemSelectionModel.SelectionFlag.ClearAndSelect
```
see doc: https://doc.qt.io/qt-6/qitemselectionmodel.html#SelectionFlag-enum

and your 

```
QItemSelectionModel.ClearAndSelect
```

Is old style enum alias. 

Ad compare it with this section https://doc.qt.io/qtforpython-6/considerations.html#doing-a-smooth-transition-from-the-old-enums from documentation that you linked.

Reply:
You are correct. in this case the current version on the `main` branch does work. I was hitting this bug in 0.4.17 (latest conda-forge release) and I didn't realize that this fix is in fact ok.

Apologies for the noise here and thank you for explaining things thoroughly.